lib/savings_and_checking.rb

```diff
@@ -0,0 +1,126 @@
+module Bank
+  class SavingsAccount < Account
+    def initialize(id, balance, open_date)
+      super(id, balance, open_date)
+        if @balance < 1000
+          raise ArgumentError, "Accounts cannot be opened with less than $10. Please try again."
+        end
+    end
+
+    def withdraw(withdraw_amount)
+      transaction_fee = 200
+      temp_balance = @balance - transaction_fee.to_i - withdraw_amount.to_i
+      if temp_balance < 1000
+        puts "You cannot have less than $10. This withdraw has been cancelled."
+      else
+        @balance = temp_balance
+        puts "You new balance is #{@balance}"
+      end
+      return @balance
+    end
+
+    def add_interest(rate)
+      interest_rate = rate.to_f
+      added_rate = @balance * (interest_rate/100.to_f)
+      @balance += added_rate
+      puts "Your balance is #{@balance}"
+      return added_rate
+    end
+  end
+
+  class CheckingAccount < Account
```

Comment:
This is good but i'd prefer if this class and the money market class were in separate files